sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -85,8 +88,30 @@ public void Initialize(ExtensionConfigContext context)
             var rule = context
                 .AddBindingRule<EventGridAttribute>()
                 //TODO - add binding for BinaryData?
-                .AddConverter<string, EventGridEvent>((str) => EventGridEvent.Parse(new BinaryData(str)))
-                .AddConverter<JObject, EventGridEvent>((jobject) =>  EventGridEvent.Parse(new BinaryData(jobject.ToString())));
+                .AddConverter<string, object>(str =>
+                {
+                    // first attempt to parse as EventGridEvent, then fallback to CloudEvent
+                    try
+                    {
+                        return EventGridEvent.Parse(new BinaryData(str));
```

Comment:
Any way we can try to parse it without throwing an exception?

Reply:
We could, but I'd rather not recreate the Parse logic here. We can potentially add TryParse methods to the EG SDK.

Reply:
Let's file an issue

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/20702